src/cpp/rtps/builtin/BuiltinProtocols.cpp

```diff
@@ -103,23 +101,26 @@ bool BuiltinProtocols::initBuiltinProtocols(
             return false;
 
         case DiscoveryProtocol_t::CLIENT:
-            mp_PDP = new PDPClient(this, allocation);
+            mp_PDP = new fastrtps::rtps::PDPClient(this, allocation);
             break;
 
         case DiscoveryProtocol_t::SERVER:
-            mp_PDP = new PDPServer(this, allocation, DurabilityKind_t::TRANSIENT_LOCAL);
+            mp_PDP = new fastdds::rtps::PDPServer2(this, allocation);
             break;
-#if HAVE_SQLITE3
-        case DiscoveryProtocol_t::BACKUP:
-            mp_PDP = new PDPServer(this, allocation, DurabilityKind_t::TRANSIENT);
-            break;
-#endif
+        /*
+         #if HAVE_SQLITE3
+                case DiscoveryProtocol_t::BACKUP:
+                    mp_PDP = new fastdds::rtps::PDPServer2(this, allocation);
+                    break;
+         #endif // if HAVE_SQLITE3
+         */
```

Comment:
Do not remove the BACKUP option.
Although the BACKUP is not ready on the DS2, it could use the DS1 instead.

Reply:
Agreed.